### PR TITLE
Switches Generated Public API Documentation from sirius to Swagger Annotations

### PR DESCRIPTION
**Pull request by sabieber**
**BREAKING:** This breaks the documentation of APIs previously documented via the `PublicService` and `ServiceParameter` annotations. (Usages in web are adapted, used in biz will be adapted after merge)

The [Swagger Annotations](https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Annotations#requestbody) for documenting APIs near code provide more functionality than the annotations we created.
So we mostly replace them with the ones provided by Swagger and implement a subset of the newly available capabilities deemed 'useful' by the author :)

We now can:
- Document the HTTP method of an API (whether to access it via POST or GET
- Provide examples for requests made to the API (for example an XML payload)
- Document multiple possible responses with status code and content types

Examples:

image

image

Fixes: SIRI-583

### CODE REVIEW

**Review by meg1502 — APPROVED** [review]

**Review by fhaScireum — APPROVED** [review]

File: src/main/java/sirius/web/services/PublicServiceInfo.java [hunk 1]
```diff
@@ -23,42 +33,105 @@ public class PublicServiceInfo {
...
+        if (numericStatus >= 400) {
+            return "orange";
+        }
+        if (numericStatus >= 300) {
+            return "blue";
+        }
+        if (numericStatus >= 200) {
+            return "green";
+        }
+        return "grey";
```
Comment (andyHa) [inline src/main/java/sirius/web/services/PublicServiceInfo.java:83]: i think we spell it "gray" in our CSS

File: src/main/java/sirius/web/services/PublicServiceInfo.java [hunk 2]
```diff
@@ -23,42 +33,105 @@ public class PublicServiceInfo {
...
+        if (httpMethod == HttpMethod.GET) {
+            return "green";
+        }
+        if (httpMethod == HttpMethod.POST) {
+            return "blue";
+        }
+        if (httpMethod == HttpMethod.HEAD) {
+            return "violet";
+        }
+        return "grey";
```
Comment (andyHa) [inline src/main/java/sirius/web/services/PublicServiceInfo.java:104]: i think we spell it "gray" in our CSS

**Review by idlira — APPROVED** [review]

File: src/main/resources/default/templates/system/api.html.pasta [hunk 3]
```diff
@@ -151,7 +151,10 @@
                                                  value="responseContents.get(0).as(io.swagger.v3.oas.annotations.media.Content.class)"/>
                                         <td>
                                             <i:if test="@isFilled(response.responseCode())">
-                                                <t:fullTag color="green" class="align-text-top mr-1">@response.responseCode()</t:fullTag>
+                                                <i:local name="responseCodeValue" value="@Value.of(response.responseCode())" />
```
Comment (mkeckmkeck) [inline src/main/resources/default/templates/system/api.html.pasta]: Unused

File: pom.xml [hunk 4]
```diff
@@ -138,6 +138,13 @@
             <artifactId>excel-streaming-reader</artifactId>
             <version>3.2.5</version>
         </dependency>
+
+        <!-- Swagger Annotations are used for API/Service documentation -->
+        <dependency>
+            <groupId>io.swagger.core.v3</groupId>
+            <artifactId>swagger-core</artifactId>
```
Comment (mkeckmkeck) [inline pom.xml:145]: What about using swagger-annotations?

**Review by mkeckmkeck — APPROVED** [review]

**Review by ymo-sci — APPROVED** [review]

**Pull request merged**